lib/dat-worker-pool/worker.rb

```diff
@@ -49,23 +50,36 @@ def raise(*args)
     def work_loop
       @on_start_callbacks.each{ |p| p.call(self) }
       loop do
-        @on_sleep_callbacks.each{ |p| p.call(self) }
-        work_item = @queue.pop
-        @on_wakeup_callbacks.each{ |p| p.call(self) }
         break if @shutdown
```

Comment:
@kellyredding - It's not entirely clear, but I moved the `break if @shutdown` to the top of the loop which fixes the issue of pulling work off the queue and then exiting because it was shutdown.